### PR TITLE
Run script from node

### PR DESCRIPTION
**Pull request by dcmwong**

### CODE REVIEW

File: src/config.json [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
   "dev": {
-    "port": 3000
+    "port": 3000,
+    "publishScriptPath" : "~/src/",
+    "publishScriptName" : "longrunning.sh"
   },
   "test": {
-    "port": 8080
+    "port": 8080,
+    "isPublishRunningScriptName" : "longrunning.sh"
```
Comment (NSmeef) [inline src/config.json]: Is this accurately named?

Comment (NSmeef) [inline src/config.json]: Its just the script thats we're going to call, pgrep is what determines that particular script is running?

File: src/lib/errorHandler.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module.exports = function(err, req, res, next) {
+    console.error(err);
```
Comment (NSmeef) [inline src/lib/errorHandler.js]: Generic for now

File: src/server.js [hunk 3]
```diff
@@ -1,10 +1,27 @@
...
+app.get('/publish',
+    publishStatus.checkStatus,
+    function(req, res) {
+        res.render("publishNotRunning", {});
+        res.end();
+    });
+
+app.post('/publish',
+    publishStatus.checkStatus,
+    publishStatus.publish,
```
Comment (NSmeef) [inline src/server.js]: Tidy

File: src/server.js [hunk 4]
```diff
@@ -1,10 +1,27 @@
 var express = require("express");
 var app = express();
-var publishStatus = require('./lib/publishStatus.js');
+var publishStatus = require('./lib/publishStatus.js')();
+var config = require('./lib/appConfig.js');
```
Comment (NSmeef) [inline src/server.js]: are we using `config`?

File: src/lib/appConfig.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+var exec = require('child_process').exec;
+var config = require('../config.js')();
+
+module.exports.setup = function(req, res, next) {
+  req.isPublishRunning = function() {
+    exec("pgrep " + config.isPublishRunningScriptName, function(err, stdout, stderr) {
```
Comment (NSmeef) [inline src/lib/appConfig.js]: Where are we using this?

File: src/config.json (hunk 1 above)
Comment (dcmwong) [inline src/config.json]: Rename.  publishSript?

File: src/lib/errorHandler.js (hunk 2 above)
Comment (dcmwong) [inline src/lib/errorHandler.js]: for now..yep

File: src/config.json (hunk 1 above)
Comment (NSmeef) [inline src/config.json]: Yeah, makes sense to me that

File: src/server.js (hunk 4 above)
Comment (dcmwong) [inline src/server.js]: Yes taking this out

**Pull request merged**